File: src/main/java/com/example/demo/auth/jwt/JwtTokenType.java
package com.example.demo.auth.jwt;

import org.springframework.security.oauth2.jwt.JwtClaimNames;

public enum JwtTokenType {
    ACCESS("access"),
    REFRESH("refresh");

    public static final String CLAIM = "token_type";
    public static final String ID_CLAIM = JwtClaimNames.JTI;

    private final String value;

    JwtTokenType(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: src/main/java/com/example/demo/auth/jwt/JwtTokenType.kt
package com.example.demo.auth.jwt

import org.springframework.security.oauth2.jwt.JwtClaimNames

enum class JwtTokenType(private val value: String) {
    ACCESS("access"),
    REFRESH("refresh");

    fun value(): String = value

    companion object {
        const val CLAIM: String = "token_type"
        const val ID_CLAIM: String = JwtClaimNames.JTI
    }
}
